src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -136,15 +136,30 @@ public CheckResult Check(string expressionText, FormulaType parameterType = null
             return CheckInternal(expressionText, parameterType, intellisense: false);
         }
 
+        /// <summary>
+        /// Type check a formula without executing it. 
+        /// </summary>
+        /// <param name="formula">The formula to check.</param>
+        /// <param name="parameterType">The type context.</param>
+        /// <remarks>Used by PVA.</remarks>
+        /// <returns></returns>
+        internal CheckResult Check(Formula formula, FormulaType parameterType = null)
```

Comment:
>Formula [](http://example.com/codeflow?start=35&length=7)

Note that we're tyring to move off internals, so Formula would need to be public too.

Reply:
I tried, it changes a lot of things from internal to public and starts to require the summaries...

Reply:
Should we go for a 2 stage approach and keep the Formula object internal?

public ParseResult {
     public TexlNode ParseTree => Formula.ParseTree;
     public ExpressionError[] Errors => Formula.GetParseErrors();
     public bool IsSuccess => Errors == null;


     private Formula _formula;
}

ParseResult Parse(string expression)

CheckResult Check(IParseResult parseResult, FormulaType parameterType = null)

Reply:
obviously also still keeping 
CheckResult Check(string expression, FormulaType parameterType)

Reply:
TexlNode is also internal that brings a lot of other internal stuff. Let me see if I can give this PR to the team to follow up change internal to public

Reply:
FYI, `TexlNode` should be public soon. See: #248.

Reply:
Oh, nice, then I'll wait for this PR to be merged :)

Reply:
We do need to huddle about the right shape for the public surface here.  Ability to parse-only makes sense, but Formula class may not be the right public surface.